src/service/url-replacements-impl.js

```diff
@@ -1116,18 +1120,27 @@ export class UrlReplacements {
       href = addParamsToUrl(href, parseQueryString(defaultUrlParams));
     }
 
-    if (whitelist) {
-      href = this.expandUrlSync(
-        href,
-        /* opt_bindings */ undefined,
-        /* opt_collectVars */ undefined,
-        /* opt_whitelist */ whitelist
-      );
-    }
+    href = this.expandSyncIfWhitelist_(href, whitelist);
```

Comment:
Do we need to expand again?

Reply:
Yes, the first expansion is for the `data-amp-addparams` if they exist. This expansion is for the `href` as a whole.

Reply:
Make sense. Thanks for the clarification.